internal/controller/thanosquery_controller.go

```diff
@@ -168,6 +170,13 @@ func (r *ThanosQueryReconciler) syncResources(ctx context.Context, query monitor
 			"resource", obj.GetName(), "namespace", obj.GetNamespace(),
 		)
 	}
+	if query.Spec.EnableSelfMonitor != nil && !*query.Spec.EnableSelfMonitor {
+		err := manifests.DeleteServiceMonitor(ctx, r.Client, query.Name, query.Namespace)
+		if err != nil {
```

Comment:
this will go into an endless loop if the object is already deleted? i think you will want to check for 'does not exist' in the error

Reply:
I do check for `errors.IsNotFound` in the function definition. Is that what you meant?